src/graph/utils/verbose.cpp

```diff
@@ -241,7 +241,75 @@ void partition_info_t::init(const engine_t *engine,
     });
 }
 
+static setting_t<uint8_t> graph_dump_modes {0};
+
+uint8_t parse_graph_dump_mode(const std::string &modes) {
+    uint8_t m = 0;
+    if (modes.empty()) return m;
+
+    std::string user_opt = modes;
+    std::transform(
+            user_opt.begin(), user_opt.end(), user_opt.begin(), ::tolower);
+    user_opt += ','; // Add a trailing comma to process the last token
+    size_t start = 0, end = 0;
+    while ((end = user_opt.find(',', start)) != std::string::npos) {
+        std::string token = user_opt.substr(start, end - start);
+        if (token == "subgraph")
+            m |= static_cast<uint8_t>(graph_dump_mode_t::subgraph);
+        else if (token == "graph")
+            m |= static_cast<uint8_t>(graph_dump_mode_t::graph);
+        else if (token == "pattern")
+            m |= static_cast<uint8_t>(graph_dump_mode_t::pattern);
+        else
+            m = static_cast<uint8_t>(graph_dump_mode_t::none);
```

Comment:
nit: Is the `static_cast` really needed?

Reply:
yes, there will be type mismatch between uint8_t and graph_dump_mode_t if I remove the cast.